mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb

```diff
@@ -1322,20 +1322,46 @@ func flattenStatefulPolicyStatefulInternalIps(statefulPolicy *compute.StatefulPo
 	return result
 }
 
-func flattenStatefulPolicyStatefulExternalIps(statefulPolicy *compute.StatefulPolicy) []map[string]interface{} {
+func flattenStatefulPolicyStatefulExternalIps(d *schema.ResourceData, statefulPolicy *compute.StatefulPolicy) []map[string]interface{} {
 	if statefulPolicy == nil || statefulPolicy.PreservedState == nil || statefulPolicy.PreservedState.ExternalIPs == nil {
-		return make([]map[string]interface{}, 0, 0)
+		return make([]map[string]interface{}, 0)
 	}
-	result := make([]map[string]interface{}, 0, len(statefulPolicy.PreservedState.ExternalIPs))
+
+	// statefulPolicy.PreservedState.ExternalIPs is affected by API-side reordering of external IPs, where ordering
+	// is done by the interface_name value.
+	// Below we intend to reorder the external IPs to match the order in the config.
+	// Also, data is converted from a map (client library's statefulPolicy.PreservedState.ExternalIPs) to a slice (stored in state).
+	// Any external IPs found from the API response that aren't in the config are appended to the end of the slice.
+
+	configIpOrder := d.Get("stateful_external_ip").([]interface{})
+	order := map[string]int{} // record map of interface name to index
+	for i, el := range configIpOrder {
+		externalIp := el.(map[string]interface{})
+		interfaceName := externalIp["interface_name"].(string)
+		order[interfaceName] = i
+	}
+
+	orderedResult := make([]map[string]interface{}, len(configIpOrder))
+	unexpectedIps := []map[string]interface{}{}
 	for interfaceName, externalIp := range statefulPolicy.PreservedState.ExternalIPs {
 		data := map[string]interface{}{
 			"interface_name": interfaceName,
-			"delete_rule": externalIp.AutoDelete,
+			"delete_rule":    externalIp.AutoDelete,
 		}
 
-		result = append(result, data)
+		index, found := order[interfaceName]
+		if !found {
+			unexpectedIps = append(unexpectedIps, data)
+			continue
+		}
+		orderedResult[index] = data // Put elements from API response in correct order
 	}
-	return result
+	if len(unexpectedIps) > 0 {
+		// Additional IPs returned from API but not in the config are appended to the end of the slice
```

Comment:
What would happen if I had [nic0, nic1] locally and got back (nic1, nic2) from the API? I don't think I saw a case covering this, but could have missed it.

Reply:
There wasn't a case for this, and it uncovered a bug! I just rebased the branch but the new changes are in 
db91a96